util/network/src/connection.rs

```diff
@@ -312,16 +312,16 @@ impl EncryptedConnection {
 		}
 		let mut key_material = H512::new();
 		shared.copy_to(&mut key_material[0..32]);
-		nonce_material.sha3_into(&mut key_material[32..64]);
-		key_material.sha3().copy_to(&mut key_material[32..64]);
-		key_material.sha3().copy_to(&mut key_material[32..64]);
+		keccak_into(&nonce_material, &mut key_material[32..64]);
```

Comment:
Is this really more ergonomic? What's wrong with using a trait here? I wouldn't have an opinion either way if this was original code but since it's the stated purpose of the PR it deserves analysis.

Reply:
It's one of the purposes :) imo, there is no point in using the trait if simple function is enough